Unify and fix rjson string and string_view conversions

nuivall opened the pull request:
This patch-set consolidates and corrects rjson string conversion handling.
It removes unnecessary string copies, ensures proper length usage and 
replaces ad-hoc conversions with consistent helper functions.

Overall, the changes make rjson string handling safer, faster, and more uniform across the codebase.

Backport: no, it's a refactor

nyh commented on alternator/executor.cc:
```diff
@@ -3364,7 +3364,7 @@ static bool hierarchy_filter(rjson::value& val, const attribute_path_map_node<T>
         rjson::value newv = rjson::empty_object();
         for (auto it = v.MemberBegin(); it != v.MemberEnd(); ++it) {
-            std::string attr = it->name.GetString();
+            std::string attr = rjson::to_sstring(it->name);
```
Again, wants std::string, not sstring, and can just use to_string_view().

nyh commented on alternator/ttl.cc:
```diff
@@ -93,7 +93,7 @@ future<executor::request_return_type> executor::update_time_to_live(client_state
         co_return api_error::validation("The length of AttributeName must be between 1 and 255");
     }
-    sstring attribute_name(v->GetString(), v->GetStringLength());
+    sstring attribute_name = rjson::to_sstring(*v);
```
This change is correct, but this line was also correct before this patch.

nuivall replied:
yes, I tried to unify, because it's easier to maintain if code is similar

nuivall commented on alternator/executor.cc:
```diff
@@ -2783,10 +2783,11 @@ static void verify_all_are_used(const rjson::value* field,
     }
     for (auto it = field->MemberBegin(); it != field->MemberEnd(); ++it) {
-        if (!used.contains(it->name.GetString())) {
+        auto field_name = rjson::to_string_view(it->name);
```
oh, no! thanks

nuivall commented on alternator/serialization.cc:
```diff
@@ -496,7 +496,7 @@ const std::pair<std::string, const rjson::value*> unwrap_set(const rjson::value&
     }
     auto it = v.MemberBegin();
-    const std::string it_key = it->name.GetString();
+    const std::string it_key = rjson::to_sstring(it->name);
```
thx

nuivall commented on alternator/executor.cc:
```diff
@@ -2362,7 +2362,7 @@ put_or_delete_item::put_or_delete_item(const rjson::value& item, schema_ptr sche
     _cells->reserve(item.MemberCount());
     for (auto it = item.MemberBegin(); it != item.MemberEnd(); ++it) {
-        bytes column_name = to_bytes(it->name.GetString());
+        bytes column_name = to_bytes(rjson::to_string_view(it->name).data());
```
retracting this change too, will just use rjson::to_string

nuivall commented on alternator/conditions.cc:
```diff
@@ -42,7 +42,7 @@ comparison_operator_type get_comparison_operator(const rjson::value& comparison_
         throw api_error::validation(fmt::format("Invalid comparison operator definition {}", rjson::print(comparison_operator)));
     }
-    std::string op = comparison_operator.GetString();
+    std::string op = rjson::to_sstring(comparison_operator);
```
changed to `std::string op = rjson::to_string(comparison_operator);`

nyh commented on alternator/executor.cc:
```diff
@@ -2783,7 +2783,7 @@ static void verify_all_are_used(const rjson::value* field,
     }
     for (auto it = field->MemberBegin(); it != field->MemberEnd(); ++it) {
-        if (!used.contains(it->name.GetString())) {
+        if (!used.contains(rjson::to_string(it->name))) {
```
Again, I'm confused.
Obviously used.contains(it->name.GetString()) worked, and didn't need to make a copy of the string. So why do we suddenly need to make a copy with `to_string`? If you use `to_string_view()` here - it doesn't work?

Or maybe even in the old code, even though we used `char*` it was implicitly converted into an std::string, so there is no performance regression here.

nuivall replied:
For consistency we shouldn't use GetString directly I think. And there is this \0 handling. I think to construct string like that: string(chars, 7) is faster than string(chars) which needs to find \0 char.

nyh replied:
I see. I was hoping that if we used here to_string_view(), then even if today C++ doesn't know (?) how to search in the map without creating a std::string, in the future maybe it will be implemented. By forcing the call to to_string() to really create the std::string, you're giving up on this future optimization opportunity.
But this is a minor point (if at all a point). Never mind.

nyh replied:
Yes, I know it doesn't work "as expected" on string_view (without any copying), but because the unpatched version used "char*" and it worked - I thought perhaps by implicitly creating a std::string from it - I was hoping it can also work the same way for std::string_view. From this discussion I guess it can't - but I didn't check.

nyh commented on alternator/executor.cc:
```diff
@@ -2786,7 +2786,7 @@ static void verify_all_are_used(const rjson::value* field,
             throw api_error::validation(
                 format("{} has spurious '{}', not used in {}",
-                    field_name, it->name.GetString(), operation));
+                    field_name, field_name, operation));
```
You forgot to fix this bug you introduced in the previous version of your patch :-( It shouldn't be field_name twice.

nuivall replied:
I didn't I removed my new variable, but forgot to remove GetString here, anyway fixed now

nuivall commented:
v3:
- added to_bytes from string_view conversion (separate commit)
- improved to_sstring comment to mention copy and json string requirment
- moved to_string_view up

nyh commented on types/types.hh:
```diff
@@ -832,6 +832,12 @@ to_bytes(bytes_view x) {
+inline
+bytes
+to_bytes(std::string_view x) {
+    return to_bytes(to_bytes_view(x));
```
Wow, the definition of to_bytes_view(string_view) is in a different header file (bytes.hh)? What a mess :-(

nyh reviewed (approved):
Thanks, very nice.

mykaul commented:
@scylladb/releng - can you check what happened to the dtest pipeline here? All failed?

yaronkaikov commented:
> > @scylladb/releng - can you check what happened to the dtest pipeline here? All failed?
> 
> dtest has a timeout of 4 hours, and we reached it since dtest for CI is now waiting for local builders, i will increase the job timeout to avoid this
> 
> @roydahan @fruch FYI

https://github.com/scylladb/scylla-dtest/pull/6467
